plutus-core/plutus-core/src/PlutusCore/Bitwise/Convert.hs

```diff
@@ -25,23 +29,62 @@ import Data.ByteString qualified as BS
 import Data.Text (pack)
 import Data.Word (Word64, Word8)
 import GHC.ByteOrder (ByteOrder (BigEndian, LittleEndian))
+import GHC.Exts (Word (W#))
+import GHC.Num.Integer (integerLog2#)
+
+{- | Note [Input length limitation for IntegerToByteString].  We make
+   `integerToByteString` fail if it is called with arguments which would cause
+   the length of the result to exceed about 8K bytes because the execution time
+   becomes difficult to predict accurately beyond this point (benchmarks on a
+   number of different machines show that the CPU time increases smoothly for
+   inputs up to about 8K then increases sharply, becoming chaotic after about
+   14K).  This restriction may be removed once a more efficient implementation
+   becomes available, which may happen when we no longer have to support GHC
+   8.10. -}
+{- NB: if we do relax the length restriction then we will need two variants of
+   integerToByteString in Plutus Core so that we can continue to support the
+   current behaviour for old scripts.-}
+integerToByteStringMaximumOutputLength :: Integer
+integerToByteStringMaximumOutputLength = 8000
```

Comment:
Maybe 8192 would be better.

Reply:
I'd prefer that indeed, because 8000 feels entirely arbitrary and 8K somehow feels less arbitrary, but it's again my craziness is talking. `8192` also gives the reader a hint of what this is about.

Reply:
> I'd prefer that indeed, because 8000 feels entirely arbitrary

Oh well, I suppose so.  I'll have to change a lot of  the conformance tests now too [goes off grumbling].